sys/net/link_layer/l2util/l2util.c

```diff
@@ -238,5 +239,52 @@ int l2util_ndp_addr_len_from_l2ao(int dev_type,
     return -ENOTSUP;
 }
 
+size_t __attribute__((weak)) board_get_eui_short(netdev_t *netdev, uint16_t *addr)
+{
+    (void) netdev;
+    (void) addr;
+    return 0;
+}
+
+size_t __attribute__((weak)) board_get_eui48(netdev_t *netdev, eui48_t *addr)
+{
+    (void) netdev;
+    (void) addr;
+    return 0;
+}
+
+size_t __attribute__((weak)) board_get_eui64(netdev_t *netdev, eui64_t *addr)
+{
+    (void) netdev;
+    (void) addr;
+    return 0;
+}
+
+void l2util_get_eui_short(netdev_t *netdev, uint16_t *addr)
```

Comment:
Couldn't we mark this function as `weak` and let it unconditionally call `luid_get_short()`? I think that could result in smaller ROM. Or am I underestimating the optimisation of the compiler?

Reply:
The idea is to fall back to `luid_get_short()` if the real addresses are exhausted / there is none available for `netdev`.

Reply:
I could make those functions `static inline` though, could help a bit.

Reply:
With LTO I expect the same result either way. But I'm personally afraid to use it, as calls to external functions are memory barriers without LTO, and stop being so with. I opened a few PRs to add explicit memory barriers for `irq_disable()` and `irq_restore()`, which should be wrapped arround the majority of cases where barriers are needed anyway. I think the are now all merged and they had no effect on the generated code without LTO. But there are likely still some bugs present that only manifest wih LTO.